### PR TITLE
Extract metric_key_prefix during NotebookProgressCallback.on_evaluate

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

This PR upgrades `NotebookProgressCallback` to detect `metric_key_prefix` when `on_evaluate` is called. Useful when users override `Trainer.evaluate` and pick a non-standard prefix for train / eval / test.

Forum link where this topic was discussed: https://discuss.huggingface.co/t/logging-training-accuracy-using-trainer-class/5524?u=lewtun

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks a lot for the fix!

**Comment by sgugger** [comment]
Mmm, looks like there is a problem in the tests though.

**Comment by lewtun** [comment]
Yes, part of the problem is coming from `CallbackHandler.on_evaluate` which does not have a `kwargs` in the signature: https://github.com/huggingface/transformers/blob/f1b938fda81d4b9e8ab435cb7f37f71c9b7cbb1e/src/transformers/trainer_callback.py#L361

Adding `kwargs` there seems to work, so my question is whether we should also add `kwargs` to the other class functions (e.g. `on_train_begin` etc)? Since `CallbackHandler` is a subclass of `TrainerCallback`, this would preserve the function signatures in the derived class

**Comment by sgugger** [comment]
This would be a breaking change for all users that have implemented their custom `TrainerCallback`. Maybe we can just not try to pass the `eval_prefix` and just look for anything that is `xxx_loss` in the metrics dictionary, then `xxx` is the `eval_prefix`?

**Comment by lewtun** [comment]
Ah good point. I've followed your suggestion instead 😃

File: src/transformers/utils/notebook.py [hunk 1]
```diff
@@ -318,13 +319,16 @@ def on_evaluate(self, args, state, control, metrics=None, **kwargs):
                 values["Epoch"] = int(state.epoch)
             else:
                 values["Step"] = state.global_step
-            values["Validation Loss"] = metrics["eval_loss"]
```
Comment (lewtun) [inline src/transformers/utils/notebook.py:321]: this line seemed to be redundant since we later loop over `metrics.items()`, so i removed it

**Review by sgugger — APPROVED** [review]
That's looking better, thanks a lot for adapting! Fingers crossed for the tests :-)

**Comment by lewtun** [comment]
It seems the torch tests timed out (not sure how my changes could induce that). Would it be possible to rerun the CI?

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @lewtun!

**Pull request merged**